IntelliTect.TestTools.Data/DatabaseFixture.cs

```diff
@@ -84,14 +90,18 @@ private TDbContext CreateNewContext()
 
             var db = (TDbContext)constructorInfo.Invoke(GetConstructorValues(constructorInfo));
 
-            if (!alreadyCreated)
+            if (alreadyCreated) return db;
+
+            await db.Database.EnsureCreatedAsync();
+
+            if (InitializeDatabase is not null)
```

Comment:
A better pattern here is to use pattern matching (pun intended). By coping it into a local variable it makes it thread-safe. As opposed to accessing the property's getter twice, which is not.
```c#
if (InitializeDatabase is { } init)
{
    await init(await CreateNewContext());
}
```